Fix `authorize: yes` and `become_method: enable`

Qalthos opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->
This passes the new play context for each task as they come in to update the context for the forked process.

Also fixes the variable that enable prompts are stored in for ios/eos

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
ansible-connection
network_cli

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
2.5
```

Qalthos committed:
Update connection play_context when socket exists

Qalthos committed:
Don't fail on connections other than network_cli

privateip commented on bin/ansible-connection:
```diff
@@ -293,6 +293,15 @@ def main():
+            pc_data = to_text(init_data)
+            try:
+                messages.extend(conn.update_play_context(pc_data))
+            except ConnectionError as exc:
```
i thought `utils/jsonrpc.py` will detect missing method and respond accordingly.  If not, we should fix it there, imho

Qalthos replied:
I think ConnectionError (with a specific, identifiable err code) _is_ the right response here, rather than some other, potentially surprising default.

privateip commented on lib/ansible/plugins/connection/network_cli.py:
```diff
@@ -134,19 +136,30 @@ def put_file(self, in_path, out_path):
         return self._local.fetch_file(in_path, out_path)
 
-    def update_play_context(self, play_context):
+    def update_play_context(self, pc_data):
```
should this perhaps be part of `ConnectionBase` as an noop method?

Qalthos replied:
I'm not sure this makes sense as a method for any other connection plugins, at least not while the method expects a unicode string of a pickled PlayContext object, rather than the object itself.

Qalthos committed:
Fix enable prompt detection on ios & eos

Qalthos committed:
Check against "Module not found" error code, defined in modules/jsonrpc.py

Not as clumsy or random as a string match. An elegant tool... for a more
civilised age.

Qalthos committed:
Expose more general errors, too

Qalthos committed:
Python 2.6 also needs this to be bytes

Qalthos committed:
I'm pretty sure this is the same thing
